utilities/transactions/pessimistic_transaction.cc

```diff
@@ -230,10 +230,15 @@ Status WriteCommittedTxn::PrepareInternal() {
   WriteOptions write_options = write_options_;
   write_options.disableWAL = false;
   WriteBatchInternal::MarkEndPrepare(GetWriteBatch()->GetWriteBatch(), name_);
+  uint64_t seq_used = kMaxSequenceNumber;
   Status s =
       db_impl_->WriteImpl(write_options, GetWriteBatch()->GetWriteBatch(),
-                          /*callback*/ nullptr, &log_number_, /*log ref*/ 0,
-                          /* disable_memtable*/ true);
+                      /*callback*/ nullptr, &log_number_, /*log ref*/ 0,
+                      /*disable_memtable*/ true, &seq_used);
+  assert(!s.ok() || seq_used != kMaxSequenceNumber);
+  if (s.ok()) {
+    SetId(seq_used);
```

Comment:
My understanding was that we want to use the commit sequence number to represent the transaction id, but here we are using the prepare sequence number!

Reply:
oh ok, the sequence number in the WAL at prepare time, is different from the sequence number used at in the memtable at commit time ?
In my case I was not using the `Prepare()` method, so I just thought this would fit as this code also wrote into the wal. I will move it into `CommitInternal()`